hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/FilterVisitor.java

```diff
@@ -242,26 +242,39 @@ public void visit(HawkularPipeline<?, ?> query, SwitchElementType filter, QueryT
         switch (filter.getDirection()) {
             case incoming:
                 if (jumpFromEdge) {
+                    state.setInEdges(false);
+                    state.setComingFrom(null);
                     query.outV();
                 } else {
+                    state.setInEdges(true);
+                    state.setComingFrom(Direction.IN);
                     query.inE();
                 }
                 break;
             case outgoing:
                 if (jumpFromEdge) {
+                    state.setInEdges(false);
+                    state.setComingFrom(null);
                     query.inV();
                 } else {
+                    state.setInEdges(true);
+                    state.setComingFrom(Direction.OUT);
                     query.outE();
                 }
                 break;
             case both:
                 if (jumpFromEdge) {
+                    state.setInEdges(false);
+                    state.setComingFrom(null);
                     query.bothV();
                 } else {
+                    state.setInEdges(true);
+                    state.setComingFrom(Direction.BOTH);
                     query.bothE();
                 }
                 break;
         }
+        state.setExplicitChange(true);
```

Comment:
This ``state#ExplicitChange`` is used in  ``static <S, E> void finishPipeline``, but I'm wondering why it always sets the ``ExplicitChange(true);`` to true.